### DISCUSSION TITLE
openapi types

### DISCUSSION THREAD

**Discussion by BrycensRanch** [sindresorhus/got#2231] [Q&A] [answered]
#### What would you like to discuss?
hey so I wanted to see if anyone using got has figured out a solution to have the response be typed from an openapi doc? 
...

#### Checklist

- [x] I have read the documentation.

**Comment by sindresorhus** [comment, marked as answer]
Got doesn't have built-in OpenAPI type generation, but you can combine it with tools like [`openapi-typescript`](https://github.com/drwpow/openapi-typescript) to get typed responses:

```typescript
import got from 'got';
import type { paths } from './generated/api'; // Generated from your OpenAPI spec

type GetUserResponse = paths['/users/{id}']['get']['responses']['200']['content']['application/json'];

const user = await got('https://api.example.com/users/123').json<GetUserResponse>();

// Now `user` is fully typed based on your OpenAPI spec
console.log(user.name);
```
